content/Video_Display.md

```diff
@@ -106,10 +107,10 @@ hidden by the text box.
 ::: warning
 As mentioned in the description of the STAT register, the PPU cycles
 through the different modes in a fixed order. If we set the STAT bits
-in a way that they would interrupt the CPU at two
-consecutive modes, then the second interrupt will not trigger. So for example,
-if we enable the interrupts for Mode 0 and Mode 1,
-the Mode 1 interrupt will not trigger.
+in a way that they would interrupt the CPU at two consecutive modes, 
```

Comment:
This is not exact; sources don't pulse an interrupt line when their respective condition becomes true, but they instead control whether to OR certain signals into the STAT int line.
This is important because it means that enabling a certain condition while it's being true *will* request a STAT interrupt (independently of the write glitch on DMG).

How should we phrase this explanation in a clear but accurate way? Perhaps move it to a different section with some diagrams?

Reply:
I'm probably not the best person to write updated text for STAT blocking since I don't have as deep an understanding of the hardware, but here's a shot anyway. :)

---
FROM:
As mentioned in the description of the STAT register, the PPU cycles through the different modes in a fixed order. If we set the STAT bits in a way that they would interrupt the CPU at two consecutive modes, then the second interrupt will not trigger. So for example, if we enable the interrupts for Mode 0 and Mode 1, the Mode 1 interrupt will not trigger.

----
TO:
The various STAT interrupt sources (modes 0-2 and LYC=LY) logical OR their state (inactive/low and active/high) into a shared STAT interrupt line if their respective mode enable bit is turned on.

A STAT interrupt will be triggered by a rising edge (transition from low-to-high) on the STAT interrupt line.

If a STAT interrupt source logical ORs the interrupt line high while (or immediately after) it's already set high by another source, then there will be no low-to-high transition and so no interrupt will occur. This phenomenon is known as "STAT blocking".

As mentioned in the description of the STAT register, the PPU cycles through the different modes in a fixed order. So for example, if interrupts are enabled for two consecutive modes such as Mode 0 and Mode 1, then no interrupt will trigger for Mode 1 (since the STAT interrupt line won't have a chance to go low between them).

Reply:
That sounds good! [Here is the test ROM that can be cited as reference.](https://github.com/Gekkio/mooneye-gb/blob/2d52008228557f9e713545e702d5b7aa233d09bb/tests/acceptance/ppu/stat_irq_blocking.s#L21-L22)